src/routes/guides/testing-your-app.mdx

```diff
@@ -0,0 +1,404 @@
+
+```tsx
+const { findByText } = render(
+  () => <Route path="/article/:id" component={Article} />,
+  { location: "/article/12345" }
+);
+expect(await findByText("Article 12345")).toBeInTheDocument()
+```
+
+### Interacting with components
+
+Many components are not static, but instead change on user interactions. These interactions need to be simulated to test the changes. To simulate user interactions, the creators of testing-library have provided the `@testing-library/user-event` library. It takes care of the usual order of events as they would occur in actual user interactions, e.g. that a `click` is accompanied by `mousemove`, `hover`, `keydown`, `focus`, `keyup`, and `keypress`.
+
+The most convenient events are usually `click`, `keyboard` and `pointer` (to simulate touch events). Learn more about them on the [user-event documentation](https://testing-library.com/docs/user-event/intro). If you want to use fake timers with `vi.useFakeTimers()` in your tests, be sure to set it up with an `advanceTimers` option.
+
+```tsx title="user-event.test.tsx"
+import { vi } from "vitest"
+
+const user = userEvent.setup({ advanceTimers: vi.advanceTimersByTime })
+
+vi.useFakeTimers()
+
+describe("pre-login: sign-in", () => {
+  const { getByRole, getByLabelText } = render(() => <User />)
+  const signUp = getByRole('button', { text: 'Sign-in' })
+  // use convenience API click:
+  user.click(signUp)
+  const name = getByLabelText('Name')
+  // use complex keyboard input:
+  user.keyboard(name, "{Shift}test{Space}{Shift}user")
+  const password = getByLabelText('Password')
+  user.keyboard(name, "secret")
+  const login = getByRole('button', { text: 'Login' })
+  // use touch event
+  user.pointer([
+    { keys: "[TouchA]" target: login },
+    { keys: "[/TouchA]", target: login }
+  ])
+});
+```
+
+### Validating assertions
+
+`vitest` comes with the `expect` function to facilitate assertions that works like:
+
+```tsx frame="none"
+expect(subject)[assertion](value)
+```
+
+The command supports assertions like `toBe` (reference comparison) and `toEqual` (value comparison) out of the box. For testing inside the DOM, the package `@testing-library/jest-dom` augments it with some helpful additional assertions:
+
+- `.toBeInTheDocument()` - checks if the element actually exists in the DOM
+- `.toBeVisible()` - checks if there is no reason the element should be hidden
+- `.toHaveTextContent(content)` - checks if the text content matches
+- `.toHaveFocus()` - checks if this is the currently focused element
+- `.toHaveAccessibleDescription(description)` - checks accessible description
+- and a [lot more](https://github.com/testing-library/jest-dom?tab=readme-ov-file#custom-matchers).
+
+
+## Directive testing
```

Comment:
```suggestion
### Directive testing
```

Reply:
This again makes no semantic sense, as directive testing is not a part of the previous h2 "testing commponents".